modal/_output.py

```diff
@@ -175,17 +175,14 @@ def __init__(
         self._app_page_url = None
         self._show_image_logs = False
 
+    def is_visible(self) -> bool:
+        return self._visible_progress
+
     def hide_output(self):
         self._visible_progress = False
 
-    def print_if_visible(self, renderable) -> None:
-        if self._visible_progress:
-            self._console.print(renderable)
-
-    def ctx_if_visible(self, context_mgr):
-        if self._visible_progress:
-            return context_mgr
-        return contextlib.nullcontext()
+    def print(self, renderable) -> None:
+        self._console.print(renderable)
```

Comment:
Would it clean things up even further to make this a no-op when `self.is_visible` is False?

Reply:
See previous – I think it will be cleaner to keep it this way once I make it a singleton, something like
```python
if OutputManager.is_enabled():
    import rich
    renderable = ...
    OutputManager.get_singleton().print(renderable)
```

something like that?